internal/commands/pull/inputs.go

```diff
@@ -147,8 +155,14 @@ func (i *inputs) resolveClientTemplates(ui terminal.UI, realmClient realm.Client
 		} else if !ok {
 			return nil, fmt.Errorf("template '%s' does not have a frontend to pull", templateID)
 		}
-		result[templateID] = templateZip
+		result = append(result, clientTemplate{
+			id:     templateID,
+			client: templateZip,
+		})
 	}
+	sort.SliceStable(result, func(i, j int) bool {
```

Comment:
I don't think this is needed ... we already sort on the backend, unless i'm missing something everything should already be deterministic, right?

Reply:
I think this is needed because `result` is populated by iterating through a map and getting the result of `realmClient.ClientTemplate` which returns a singular payload.

The sorted output on the backend is the output of the endpoint that returns (templateId, templateName) pairs that are used to display as options to choose a template from.

Reply:
how is `result` populated by iterating through a `map`?

`nameOptions` is built by iterating through `CompatiableTemplates` (which is an array) and `selectedTemplateIdxs` is also an array.  i'm probably missing something?

Reply:
ah right, that's true. I saw the for-loop and for some reason assumed it was a map. I'll remove